build_tools/cmake/iree_llvm.cmake

```diff
@@ -82,9 +86,6 @@ macro(iree_llvm_configure_installed)
   list(APPEND CMAKE_MODULE_PATH "${LLD_CMAKE_DIR}")
   include_directories(${LLD_INCLUDE_DIRS})
 
-  find_package(Clang REQUIRED)
-  list(APPEND CMAKE_MODULE_PATH "${CLANG_CMAKE_DIR}")
-
```

Comment:
I had to remove those two lines to get this to run on my machine. It was failing at FindClang like pasted there: https://github.com/openxla/iree/issues/13957#issuecomment-1579116706

The code below didn't seem to be making use of that, so I thought we might just delete these lines (?!). Caveat reviewer!